packages/rest/package.json

```diff
@@ -31,11 +31,12 @@
     "dist/**",
     "!dist/test/**"
   ],
-  "dependencies": {
+  "peerDependencies": {
     "@cadl-lang/compiler": "0.23.0"
   },
   "devDependencies": {
     "@types/node": "~14.0.27",
+    "@cadl-lang/compiler": "0.23.0",
```

Comment:
This one is a bit annoying. Peer dependencies works just fine for linking the package but rush doesn't mark peerDepdenency as a requirement for the build order so you endup with openapi3 pacakge building before compiler which makes the build fail.

There is this issue https://github.com/microsoft/rushstack/issues/2807 and I feel like the fact that there is a lock file means the need for devDependencies is useless.

Reply:
Does `strictPeerDependencies` help? https://rushjs.io/pages/maintainer/recommended_settings/

Reply:
Did see that setting but not sure if the doc is out of date or what but I get an error when using it 
![image](https://user-images.githubusercontent.com/1031227/142043981-3848cc01-83b7-4a1b-95a3-b854392ac2cc.png)
![image](https://user-images.githubusercontent.com/1031227/142043993-9712419e-7e87-40f1-99be-79fa003d714d.png)

I think also it solve the issue that if you are consuming a package that requires a peer dependency  it force you to define it.

Reply:
nvm its under `pnpmOptions`

Reply:
didn't solve the issue anyway

Reply:
wonder if this is something npm workspace would deal with better.